### DISCUSSION TITLE
FerretDB docker image size

### DISCUSSION THREAD

**Discussion by milangrahovac** [FerretDB/FerretDB#1767] [General]
Hello!

I'm trying to play with a FerretDB docker container to see how it works and what it has to offer, and I noticed that even though the image in the registry is ˜300MB, the layers on my local env take ˜ 1GB.

I guess this is not an expected behaviour as 1GB looks a little bit too big. Does it?

I use arm docker image as I install it on raspberry pi 3B+ with the portainer.io app.

I attach a screenshot that shows the layers I see in the portainer interface:
![Screenshot 2023-01-07 at 19 03 30](https://user-images.githubusercontent.com/15969033/211164397-bc10fec1-88cc-468e-b399-c4f5a3b695f6.png)

Is this expected?

**Comment by AlekSi** [comment]
Hi Milan! Sorry for the long delay.

Currently, we ship a development version of FerretDB with all debugging facilities enabled. They include [the Go race detector](https://go.dev/doc/articles/race_detector) that requires glibc and the whole Linux distribution. That's why the current image is huge.

We plan to separate development and production images before GA. We track that as issue #323.

**Comment by AlekSi** [comment]
Since 0.9.3, our production images are much smaller than our development images:
* https://hub.docker.com/layers/ferretdb/ferretdb/0.9.3/images/sha256-2f4fc48111bac6a314be1419996ca29ba2381eea6f87e7b663a5726fc2d95649
* https://hub.docker.com/layers/ferretdb/ferretdb-dev/0.9.3/images/sha256-0b1c96e085ac47fda2701e96fd4f151f71f9b26e5bcae7d21365e6b8855d08aa
